### ISSUE TITLE
Port to Hyper 1.0

### ISSUE THREAD

**Issue by slinkydeveloper** [restatedev/restate#96]
```[tasklist]
### Tasks
- [ ] https://github.com/restatedev/restate/issues/100
- [x] Wait for hyper 1.0 to be released
- [x] Wait for hyper-rustls to support hyper 1.0
- [ ] https://github.com/restatedev/restate/issues/1060
- [ ] https://github.com/restatedev/restate/issues/1727
- [ ] https://github.com/restatedev/restate/issues/1059
- [ ] https://github.com/restatedev/restate/issues/1729
- [ ] https://github.com/restatedev/restate/issues/1730
- [ ] https://github.com/restatedev/restate/issues/1731
- [ ] https://github.com/restatedev/restate/issues/1732
- [ ] https://github.com/restatedev/restate/issues/1737
- [ ] https://github.com/restatedev/restate/issues/1745
```

**Comment by tillrohrmann** [comment]
In my latest performance measurements, I noticed that the main bottleneck seems to be the invoker currently (the ingress is also causing high CPU load but does not seem be the limiting factor). One thing I noticed in the flamegraphs is that hyper requires a lot of CPU cycles when polling the `Body`. I haven't found the reason for this problem but it could be that there is some contention on the single TCP connection that the invoker currently uses when talking to the service deployment.

I hope that by upgrading to Hyper 1.1 we will manage to remove the bottleneck from the server.
